Src/FluentAssertions/Equivalency/Steps/DataRowEquivalencyStep.cs

```diff
@@ -205,15 +205,34 @@ private class SelectedDataRowMembers
                     new MemberSelectionContext(comparands.CompileTimeType, comparands.RuntimeType, config));
             }
 
-            selectedMembers = new SelectedDataRowMembers
-            {
-                HasErrors = members.Any(m => m.Name == nameof(DataRow.HasErrors)),
-                RowState = members.Any(m => m.Name == nameof(DataRow.RowState))
-            };
+            selectedMembers = GetSelectedDataRowMembers(members);
```

Comment:
Wouldn't a simple `ToArray()` not be enough?

Reply:
That forces a _complete_ enumeration of `members` where as this uses _at most_ a single enumeration, but I agree with your preference in using `ToArray`.

Reply:
I reckon that's negligible with the kinds of assertions people would be doing.